src/Compilers/CSharp/Portable/Compiler/DocumentationCommentCompiler.cs

```diff
@@ -403,10 +406,73 @@ private static bool ShouldSkip(Symbol symbol)
         {
             return symbol.IsImplicitlyDeclared ||
                 symbol.IsAccessor() ||
-                symbol is SynthesizedSimpleProgramEntryPointSymbol ||
-                symbol is SynthesizedRecordPropertySymbol;
+                symbol is SynthesizedSimpleProgramEntryPointSymbol;
+        }
+
+        private bool TryProcessRecordPropertyDocumentation(
+            SynthesizedRecordPropertySymbol recordPropertySymbol,
+            ImmutableArray<DocumentationCommentTriviaSyntax> docCommentNodes,
+            [NotNullWhen(true)] out string? withUnprocessedIncludes,
+            out ImmutableArray<CSharpSyntaxNode> includeElementNodes)
+        {
+            _cancellationToken.ThrowIfCancellationRequested();
+
+            if (getMatchingParamTags() is not { } paramTags)
+            {
+                withUnprocessedIncludes = null;
+                includeElementNodes = default;
+                return false;
+            }
+
+            Debug.Assert(paramTags.Count > 0);
+
+            BeginTemporaryString();
+            WriteLine("<member name=\"{0}\">", recordPropertySymbol.GetDocumentationCommentId());
+            Indent();
+            var substitutedTextBuilder = PooledStringBuilder.GetInstance();
+            var includeElementNodesBuilder = _processIncludes ? ArrayBuilder<CSharpSyntaxNode>.GetInstance() : null;
+            DocumentationCommentWalker.GetSubstitutedText(_compilation, recordPropertySymbol, paramTags, includeElementNodesBuilder, substitutedTextBuilder.Builder);
+            string substitutedText = substitutedTextBuilder.ToStringAndFree();
+            string formattedXml = FormatComment(substitutedText);
+            Write(formattedXml);
+            Unindent();
+            WriteLine("</member>");
+
+            withUnprocessedIncludes = GetAndEndTemporaryString();
+            includeElementNodes = includeElementNodesBuilder?.ToImmutableAndFree() ?? default;
+
+            paramTags.Free();
+            return true;
+
+            ArrayBuilder<XmlElementSyntax>? getMatchingParamTags()
+            {
+                ArrayBuilder<XmlElementSyntax>? result = null;
+                foreach (var trivia in docCommentNodes)
+                {
+                    foreach (var contentItem in trivia.Content)
+                    {
+                        if (contentItem is XmlElementSyntax elementSyntax)
+                        {
+                            foreach (var attribute in elementSyntax.StartTag.Attributes)
+                            {
+                                if (attribute is XmlNameAttributeSyntax nameAttribute
+                                    && nameAttribute.GetElementKind() == XmlNameAttributeElementKind.Parameter
+                                    && string.Equals(nameAttribute.Identifier.Identifier.ValueText, recordPropertySymbol.Name, StringComparison.Ordinal))
```

Comment:
We could consider actually binding the attribute value and comparing the resulting symbol with the record property symbol, but it doesn't seem necessary.